bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoTemperatureSensorHandler.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.neohub.internal;
+
+import static org.openhab.binding.neohub.internal.NeoHubBindingConstants.*;
+
+import javax.measure.Unit;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Thing;
+
+/**
+ * The {@link NeoTemperatureSensorHandler} is the OpenHAB Handler for NeoTemperatureSensor devices
+ * 
+ * Note: inherits almost all the functionality of a {@link NeoBaseHandler}
+ * 
+ * @author Andrew Fiddian-Green - Initial contribution
+ * 
+ */
+public class NeoTemperatureSensorHandler extends NeoBaseHandler {
```

Comment:
Please add `@NonNullByDefault`

Reply:
I feared you might say that @cpmeister :( — I think the logical conclusion would be for me to retroactively do that to all of the classes in the binding, and not only just the new classes. Do you guys have an opinion on this?

Reply:
It is not my intent to make a PR any more complicated than it needs to be. I only want you to add `@NonNullByDefault` to new classes that are added to the binding, the existing classes can be considered to be grandfathered in.

Of course, nothing is stopping you from adding them in if you wish but I would prefer if that were a separate PR. This PR already has too much going on.

Reply:
@cpmeister hmm, these two new classes are trivial inherited classes from the base handler class; so if I add `@NonNullByDefault` to them, it causes an mvn build error by differing from the parent class. 

To correct the mvn build errors, I would need to add `@NonNullByDefault` to the parent base class. Which means I would have to do the same to the other classes derived from the base. And I suspect this would cause a full rat's tail of null analysis changes to many other classes in the binding.

So, in fact I would prefer to follow your prior suggestion, and defer ALL of the null annotation changes to a separate PR. => Is that OK with you?

Reply:
That sounds fine by me, I don't want to add a ton of work to this PR.